### ISSUE TITLE
func-name-mixed-case clashes with snake case for constant vars

### ISSUE THREAD

**Issue by gitpusha** [protofire/solhint#268]
I guess for `constant` and `immutable` state vars which should be `SNAKE_CASE` this rule clashes for their pure getter fns:

![image](https://user-images.githubusercontent.com/36712489/97478366-1f1d6e80-1951-11eb-86cf-875c280a48fb.png)

**Comment by fvictorio** [comment]
Hey @gitpusha, could you expand on this? I'm not sure I understand the problem.

**Comment by gitpusha** [comment]
If for example you have a contract with a constant state variable like so 

```solidity 
address public constant DAI = "0x...";
```

If you want to write an interface for this contract you would then write 

```solidity 
function DAI() external pure returns (bytes4) 
```

and this would clash with that mixedCase rule 

But don't worry about it 

I am closing this issue cuz actually I prefer disabling the linter explicitly here anyhow
